reana-ui/src/actions.js

```diff
@@ -157,51 +165,29 @@ export function userSignout() {
   };
 }
 
-export function requestToken() {
-  return async dispatch => {
-    let resp, data;
-    try {
-      dispatch({ type: USER_REQUEST_TOKEN });
-      resp = await fetch(USER_REQUEST_TOKEN_URL, {
-        method: "PUT",
-        credentials: "include"
-      });
-    } catch (err) {
-      throw new Error(USER_INFO_URL, 0, err);
-    }
-    if (resp.status === 401) {
-      data = await resp.json();
-      console.log(data.message);
-    } else if (resp.ok) {
-      data = await resp.json();
-    }
-    dispatch({ type: USER_TOKEN_REQUESTED, ...data });
-    return resp;
-  };
-}
+export const requestToken = () => async (dispatch) => {
```

Comment:
Thanks a lot for the feedback! I was checking the project code and got an impression that there is a switch from using `fetch` and traditional functions to `axios` + arrow functions in the newer code. As an example I checked `GitLabProjects` component. So that's why I started using these two in the updated code. But I understand now that it's not like that :) So in this case I totally agree that we should keep the consistency. I can refactor `actions` to traditional functions

Reply:
At least for me arrow functions are more readable and sometimes can save some extra lines. Like in this case:

```
export function loadConfig() {
  return async dispatch => {
    dispatch({ type: CONFIG_FETCH });
    return await axios.get(CONFIG_URL, { withCredentials: true })
      .then(resp => dispatch({ type: CONFIG_RECEIVED, ...resp.data }))
      .catch(err => {
        dispatch(getError(err, CONFIG_URL));
        dispatch({ type: CONFIG_ERROR });
      });
  }
};
```

```
export const loadConfig = () => async (dispatch) => {
  dispatch({ type: CONFIG_FETCH });
  return await axios.get(CONFIG_URL, { withCredentials: true })
    .then(resp => dispatch({ type: CONFIG_RECEIVED, ...resp.data }))
    .catch(err => {
      dispatch(getError(err, CONFIG_URL));
      dispatch({ type: CONFIG_ERROR });
    });
};
```

Reply:
I agree, maybe what we can do is keep it as it mainly is right now and discuss this later with @mvidalgarcia, as long as we keep consistency I am fine with both.

Reply:
- WRT arrow functions vs regular functions: I don't have a strong preference. I think it's ok to use arrow functions as long as we don't deal with `this`. But I would need to read more about it.
- WRT `axios` vs `fetch`: I think `axios` is definitely more powerful. OTOH `fetch` comes with the Web API so we save a dependency and our app is lighter. The idea was finding pros and cons in [this issue](https://github.com/reanahub/reana-ui/issues/46) and centralize all the calls to the REST API in a single file (possibly `client.js`). This way we can encapsulate all the complexity in this file (no matter if we use `fetch` or `axios`!), perhaps create a class and perform the calls something like `Client.getWorkflows()`. What do you think?

Reply:
I think it's a good idea to separate api calls from redux actions, but having it in a single file could bloat it very quickly in the future. Perhaps it could be divided into some logical units (maybe similar how reana-server is divided - workflows, users, gitlab) and placed in `/api` folder?

Reply:
Definitely, we can split it as it grows.